lib/srv/desktop/rdp/rdpclient/src/rdpdr/mod.rs

```diff
@@ -655,11 +831,738 @@ impl DeviceControlResponse {
+    // is provided below for documentation purposes.
+    //
+    // Length (4 bytes): A 32-bit unsigned integer that specifies the number of bytes in the QueryBuffer field.
+    //
+    // Padding (24 bytes): An array of 24 bytes. This field is unused and MUST be ignored.
+    //
+    // QueryBuffer (variable): A variable-length array of bytes. The size of the array is specified by the Length field.
+    // The content of this field is based on the value of the FsInformationClass field, which determines the different
+    // structures that MUST be contained in the QueryBuffer field. For a complete list of these structures, see [MS-FSCC]
+    // section 2.4. The "File information class" table defines all the possible values for the FsInformationClass field.
+}
+
+#[allow(dead_code)]
+impl ServerDriveQueryInformationRequest {
+    fn decode(device_io_request: DeviceIoRequest, payload: &mut Payload) -> RdpResult<Self> {
+        if let Some(fs_information_class_lvl) =
+            FsInformationClassLevel::from_u32(payload.read_u32::<LittleEndian>()?)
+        {
+            Ok(Self {
+                device_io_request,
+                fs_information_class_lvl,
+            })
+        } else {
+            Err(invalid_data_error(
+                "received invalid FsInformationClass in ServerDriveQueryInformationRequest",
+            ))
+        }
+    }
+}
+
+/// 2.4 File Information Classes [MS-FSCC]
+/// https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-fscc/4718fc40-e539-4014-8e33-b675af74e3e1
+#[derive(Debug)]
+#[allow(dead_code, clippy::enum_variant_names)]
+enum FsInformationClass {
+    FileBasicInformation(FileBasicInformation),
+    FileStandardInformation(FileStandardInformation),
+    FileBothDirectoryInformation(FileBothDirectoryInformation),
+}
+
+#[allow(dead_code)]
+impl FsInformationClass {
+    fn encode(&self) -> RdpResult<Vec<u8>> {
+        match self {
+            Self::FileBasicInformation(file_basic_info) => file_basic_info.encode(),
+            Self::FileStandardInformation(file_standard_info) => file_standard_info.encode(),
+            Self::FileBothDirectoryInformation(fil_both_dir_info) => fil_both_dir_info.encode(), // TODO(isaiah)
+        }
+    }
+}
+
+/// 2.4.7 FileBasicInformation [MS-FSCC]
+/// https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-fscc/16023025-8a78-492f-8b96-c873b042ac50
+#[derive(Debug)]
+struct FileBasicInformation {
+    creation_time: i64,
+    last_access_time: i64,
+    last_write_time: i64,
+    change_time: i64,
+    file_attributes: flags::FileAttributes,
+    // NOTE: The `reserved` field in the spec MUST not be serialized and sent over RDP, or it will break the server implementation.
+    // FreeRDP does the same: https://github.com/FreeRDP/FreeRDP/blob/1adb263813ca2e76a893ef729a04db8f94b5d757/channels/drive/client/drive_file.c#L508
+    //reserved: u32,
+}
+
+#[allow(dead_code)]
+/// 4 i64's and 1 u32's = (4 * 8) + 4
```

Comment:
```suggestion
/// 4 i64 and 1 u32 = (4 * 8) + 4
```

I'm a stickler for misused `'`

Reply:
Uh oh, you've got a lot more of those coming your way 🥲